images/installer-artifacts/Dockerfile.rhel

```diff
@@ -1,11 +1,16 @@
 # This Dockerfile builds an image containing the Mac version of the installer layered
 # on top of the Linux installer image.
 
-FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.15-openshift-4.8 AS builder
+FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.16-openshift-4.8 AS macbuilder
 WORKDIR /go/src/github.com/openshift/installer
 COPY . .
-RUN go generate ./data && \
-    SKIP_GENERATION=y GOOS=darwin GOARCH=amd64 hack/build.sh
+RUN make cross-build-darwin-amd64
+
+FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.16-openshift-4.8 AS linuxbuilder
+WORKDIR /go/src/github.com/openshift/installer
+COPY . .
+RUN make cross-build-linux-amd64
 
 FROM registry.ci.openshift.org/ocp/4.8:installer
-COPY --from=builder /go/src/github.com/openshift/installer/bin/openshift-install /usr/share/openshift/mac/openshift-install
+COPY --from=macbuilder /go/src/github.com/openshift/installer/_output/bin/darwin_amd64/openshift-install /usr/share/openshift/mac/openshift-install
+COPY --from=linuxbuilder /go/src/github.com/openshift/installer/_output/bin/linux_amd64/openshift-install /usr/share/openshift/linux_amd64/openshift-install
```

Comment:
Help me better understand this. Is ART replacing `registry.ci.openshift.org/ocp/4.8:installer` with architecture-specific images based on the target architecture of the OpenShift release image?
Depending on the architecture of the release image, this binary may be a duplicate of the one from `registry.ci.openshift.org/ocp/4.8:installer`, right? But we want to include the duplicate
1. so that `oc` has a consistent way of extracting the installer binary for the user's desired architecture and
2. because there is no good way to use a link instead since the Dockerfile does not know which architecture ART is going to use when replacing `registry.ci.openshift.org/ocp/4.8:installer`.

Reply:
That pretty well describes it.